apps/data-pipeline/websoc-scraper/src/lib.ts

```diff
@@ -830,16 +830,9 @@ export async function scrapeTerm(
   } else {
     console.log("Performing chunk-wise scrape.");
     for (let i = 0; i < sectionCodeBounds.length; i += 2) {
-      const lower = sectionCodeBounds[i];
-      const upper = sectionCodeBounds[i + 1] ?? LAST_SECTION_CODE;
-      const sectionCodes = `${lower}-${upper}`;
-      console.log(`Scraping chunk ${sectionCodes}`);
-      const resp = await request(term, {
-        sectionCodes,
-        cancelledCourses: "Include",
-      }).then(normalizeResponse);
-      if (resp.schools.length) await doChunkUpsert(db, term, resp, null);
-      await sleep(1000);
+      const lower = sectionCodeBounds[i] as `${number}`;
```

Comment:
What is the purpose of this cast?

Reply:
Wanted to assert this is a valid int.

Reply:
As in, assert that it is a string containing a valid int?

…I guess, but this feels unnecessary. If it's because `ingestChunk` is exported and thus should have sensible inputs, it also doesn't need to be exposed.

Reply:
Without a type assertion, I cannot define the parameter types for `ingestChunk`. The use of raw SQL results in incorrect type inference.

If you want a less contrived typehint, `string` may be acceptable.

`ingestChunk` will not be exported.

Reply:
Makes sense to me now - once `ingestChunk` is unexported then lgtm